Mathlib/CategoryTheory/Limits/Shapes/ConcreteCategory.lean

```diff
@@ -0,0 +1,100 @@
+/-
+Copyright (c) 2023 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+import Mathlib.CategoryTheory.Limits.Preserves.Shapes.Products
+import Mathlib.CategoryTheory.Limits.Shapes.Types
+
+/-!
+# Limits in concrete categories
+
+In this file, we combine the description of limits in `Types` and the API about
+the preservation of products and pullbacks in order to describe these limits in a
+concrete category `C`.
+
+If `F : J → C` is a family of objects in `C`, we define a bijection
+`Limits.Concrete.productEquiv F : (forget C).obj (∏ F) ≃ ∀ j, F j`.
+
+Similarly, if `f₁ : X₁ ⟶ S` and `f₂ : X₂ ⟶ S` are two morphisms, the elements
+in `pullback f₁ f₂` are identified by `Limits.Concrete.pullbackEquiv``
+to compatible tuples of elements in `X₁ × X₂`.
+
+-/
+
+universe w v u
+
+namespace CategoryTheory
+
+namespace Limits
+
+namespace Concrete
+
+attribute [local instance] ConcreteCategory.funLike ConcreteCategory.hasCoeToSort
+
+variable {C : Type u} [Category.{v} C]
+
+section Products
+
+variable [ConcreteCategory.{max w v} C] {J : Type w} (F : J → C)
+  [HasProduct F] [PreservesLimit (Discrete.functor F) (forget C)]
+
+/-- The equivalence `(forget C).obj (∏ F) ≃ ∀ j, F j` if `F : J → C` is a family of objects
+in a concrete category `C`. -/
+noncomputable def productEquiv : (forget C).obj (∏ F) ≃ ∀ j, F j :=
+  ((PreservesProduct.iso (forget C) F) ≪≫ (Types.productIso.{w, v} (fun j => F j))).toEquiv
```

Comment:
```suggestion
  ((PreservesProduct.iso (forget C) F) ≪≫ (Types.productIso.{w, v} F)).toEquiv
```
(untested)

Reply:
Had to revert this because here `F j` is the coercion to `Type` of `F j`.